--- a/bin/debci-localtest
+++ b/bin/debci-localtest
@@ -9,7 +9,7 @@
 
 pkg="$1"
 shift
-logs="$(mktemp --directory --tmpdir debci-localtest.XXXXXXXXXX)"
+logs="${debci_logs_dir:-$(mktemp --directory --tmpdir debci-localtest.XXXXXXXXXX)}"
 
 # these variables can be considered as an API by backends/*/test-package and
 # debci-autopkgtest
@@ -19,12 +19,8 @@
 export debci_test_package="$pkg"
 export debci_test_outputdir="$logs"
 
-extra_opts=
-if [ -t 1 ]; then
-  extra_opts=--shell-fail
-fi
 rc=0
-test-package --output-dir="$logs" $extra_opts "$pkg" "$@" || rc="$?"
+test-package --output-dir="$logs" "$pkg" "$@" || rc="$?"
 
 echo "I: test artifacts available at $logs"
 exit "$rc"
